### PR TITLE
Ben/privitization

### PR DESCRIPTION
**Pull request by Ben-PH**
The drawback of using workspaces, is that the rust-tooling cannot reason about which public items are unused within the workspace.

I used a script to make every `pub ...` item `pub(crate) ...` and then cleaned up the errors by removing the `(crate)` on things that are actually needed by other workspace libraries.

Lessons learned:
- we have a lot of unused items
- we have some things that can be crate-private, but also used in doc-comments
- We need to think about preventing this bloat from happening in the future auto-magically. [if these trends continue...](https://www.youtube.com/watch?v=e6LOWKVq5sQ)
- Some things are used in another library, but only as unit-tests. I consider this a smell.

* [ ] document all added functions
* [ ] try in sandbox /simulation/labnet
* [ ] unit tests on the added/changed features
  * [ ] make tests compile
  * [ ] make tests pass 
* [ ] add logs allowing easy debugging in case the changes caused problems
* [ ] if the API has changed, update the API specification

### CODE REVIEW

**Review by AurelienFT — COMMENTED** [review]
Hasn't look much in details for now but the structures exposed in `exports` modules should remain pub even if they are not used yet. The goal of the `exports` module is to have interface that can be used by other modules. If some aren't used by anyone then maybe worth deleting them but we should never have a `pub(crate)` in `exports` module.

@damip if this kind of PR is merged the changes on consensus could affect the audit ?

**Comment by Ben-PH** [comment]
I brought up the idea of workspace-aware visibility to the core team, and seems to be getting some eyes: https://rust-lang.zulipchat.com/#narrow/stream/213817-t-lang/topic/workspace-aware.20visibility.3F

File: massa-execution-exports/src/types.rs [hunk 1]
```diff
@@ -43,7 +43,7 @@ pub enum SlotExecutionOutput {
     ExecutedSlot(ExecutionOutput),
 
     /// Finalized slot output
-    FinalizedSlot(Slot),
+    FinalizedSlot(ExecutionOutput),
```
Comment (Ben-PH) [inline massa-execution-exports/src/types.rs:46]: @aoudiamoncef I'm not too sure if this is the right thing to happen or not: something got mangled along the way when fixing errors.

File: massa-execution-worker/src/execution.rs [hunk 2]
```diff
@@ -1294,7 +1294,7 @@ impl ExecutionState {
 
         // Broadcast a final slot execution output to active channel subscribers.
         if self.config.broadcast_enabled {
-            let slot_exec_out = SlotExecutionOutput::FinalizedSlot(exec_out.slot);
+            let slot_exec_out = SlotExecutionOutput::FinalizedSlot(exec_out.clone());
```
Comment (Ben-PH) [inline massa-execution-worker/src/execution.rs:1297]: @aoudiamoncef related to privious ping

File: massa-models/src/config/constants.rs [hunk 3]
```diff
@@ -4,12 +4,10 @@
 //! # Default hard-coded
 //!
 //! Each crates may contains a `settings.rs` or a `config.rs` the `Default`
-//! implementation of each object take the default Values from the following
+//! imppubct take the default Values from the following
```
Comment (Ben-PH) [inline massa-models/src/config/constants.rs:7]: revert me

File: massa-models/src/config/constants.rs [hunk 4]
```diff
@@ -55,18 +53,18 @@ lazy_static::lazy_static! {
...
     /// TESTNET: time when the blockclique is ended.
-    pub static ref END_TIMESTAMP: Option<MassaTime> = if cfg!(feature = "sandbox") {
+    pub  static ref END_TIMESTAMP: Option<MassaTime> = if cfg!(feature = "sandbox") {
         None
     } else {
         Some(1685556000000.into())  // Sunday, May 30, 2023 06:00:00 PM UTC
     };
     /// `KeyPair` to sign genesis blocks.
-    pub static ref GENESIS_KEY: KeyPair = KeyPair::from_str("S1UxdCJv5ckDK8z87E5Jq5fEfSVLi2cTHgtpfZy7iURs3KpPns8")
+    pub  static ref GENESIS_KEY: KeyPair = KeyPair::from_str("S1UxdCJv5ckDK8z87E5Jq5fEfSVLi2cTHgtpfZy7iURs3KpPns8")
```
Comment (Ben-PH) [inline massa-models/src/config/constants.rs:62]: format me

File: massa-proto/src/_includes.rs [hunk 5]
```diff
@@ -1,14 +1,14 @@
-pub mod google {
-    pub mod api {
+pub(crate)  mod google {
+    pub(crate)  mod api {
```
Comment (Ben-PH) [inline massa-proto/src/_includes.rs:2]: confirm with @aoudiamoncef that this is valid/desired

File: massa-execution-exports/src/types.rs (hunk 1 above)
Comment (Ben-PH) [reply massa-execution-exports/src/types.rs]: This has been reverted

File: massa-execution-worker/src/execution.rs (hunk 2 above)
Comment (Ben-PH) [reply massa-execution-worker/src/execution.rs]: This has been reverted

File: massa-proto/src/massa.api.v1.rs [hunk 6]
```diff
@@ -418,23 +418,12 @@ impl BlockStatus {
...
-        /// Executed final slot
-        #[prost(message, tag = "2")]
-        FinalExecutionOutput(super::FinalizedExecutionOutput),
-    }
+    /// Status
+    #[prost(enumeration = "ExecutionOutputStatus", repeated, tag = "1")]
+    pub status: ::prost::alloc::vec::Vec<i32>,
+    /// Executed slot output
+    #[prost(message, optional, tag = "2")]
+    pub execution_output: ::core::option::Option<ExecutionOutput>,
```
[the remark this answers is not in the document]
Comment (Ben-PH) [reply massa-proto/src/massa.api.v1.rs]: this has been reverted

**Pull request closed without merging**